Submitted By: Pierre Labastie <pierre dot labastie at neuf dot fr>
Date: 2017-06-28
Initial Package Version: 2.8.8rel.2
Upstream Status: Fixed in master (at lynx-2.8.9dev;10)
Origin: Upstream
Description: Fix for building with openssl-1.1
T--- ./WWW/Library/Implementation/HTTP.c.dist    Thu Oct 27 20:10:54 2016
+++ ./WWW/Library/Implementation/HTTP.c Thu Oct 27 20:38:01 2016
@@ -720,7 +720,11 @@
 #elif SSLEAY_VERSION_NUMBER >= 0x0900
 #ifndef USE_NSS_COMPAT_INCL
 	if (!try_tls) {
-	    handle->options |= SSL_OP_NO_TLSv1;
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+           SSL_set_min_proto_version(handle,TLS1_VERSION);
+#else
+           SSL_set_options(handle, SSL_OP_NO_TLSv1);
+#endif
 #if OPENSSL_VERSION_NUMBER >= 0x0090806fL && !defined(OPENSSL_NO_TLSEXT)
 	} else {
 	    int ret = (int) SSL_set_tlsext_host_name(handle, ssl_host);
